### PR TITLE
Refactor BaseMigrationCommand to be easy to extend.

### PR DESCRIPTION
**Pull request by BernardoSilva**
Just some tidy up on code to make it easier to extend.

### CODE REVIEW

File: src/Mongrate/Command/BaseMigrationCommand.php [hunk 1]
```diff
@@ -26,20 +26,52 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-        $file = $this->getMigrationClassFileFromName($this->migrationName);
+        $this->loadMigrationClass($this->migrationName);
+    }
+
+    /**
+     * @param $className
+     *
+     * @return string
+     */
+    protected function generateFullClassName($className)
```
Comment (amyboyd) [inline src/Mongrate/Command/BaseMigrationCommand.php]: This method is only used privately so should be `private`.

File: src/Mongrate/Command/BaseMigrationCommand.php [hunk 2]
```diff
@@ -26,20 +26,52 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+    {
+        return 'Mongrate\Migrations\\' . $className;
+    }
+
+    /**
+     * @param $migrationName
+     */
+    protected function setMigrationName(Name $migrationName)
+    {
+
```
Comment (amyboyd) [inline src/Mongrate/Command/BaseMigrationCommand.php]: Extra line break here.

File: src/Mongrate/Command/BaseMigrationCommand.php [hunk 3]
```diff
@@ -26,20 +26,52 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+    }
+
+    /**
+     * Loads the migration class using the className.
+     *
+     * @param $className
+     *
+     * @throws MigrationDoesntExist
+     */
+    protected function loadMigrationClass($className)
```
Comment (amyboyd) [inline src/Mongrate/Command/BaseMigrationCommand.php]: Looks like this `$className` is a `Name` object, so maybe rename to `$name` for consistency and add type-hinting as a `Name`?

File: src/Mongrate/Command/BaseMigrationCommand.php [hunk 4]
```diff
@@ -72,4 +103,16 @@ private function setMigrationApplied($isApplied)
         $newObj = ['$set' => ['className' => (string) $this->migrationName, 'isApplied' => $isApplied]];
         $collection->upsert($criteria, $newObj);
     }
+
+    /**
+     * @param OutputInterface $output
+     *
+     * @return $this
+     */
+    protected function setOutput(OutputInterface $output)
```
Comment (amyboyd) [inline src/Mongrate/Command/BaseMigrationCommand.php]: `$this->output` is already set in `execute()` and I don't think it's needed for #34

Comment (BernardoSilva) [inline src/Mongrate/Command/BaseMigrationCommand.php]: Being used in `execute()`.
Some commands might not use parent `execute()` like  the next one.

**Comment by amyboyd** [comment]
+1 once the build passes - feel free to merge.

**Comment by BernardoSilva** [comment]
Everything passing. Looks like was travis funny behaviour before.

**Pull request merged**